creation/error-prone-actor-dependencies.md

```diff
@@ -0,0 +1,128 @@
+# Error Prone Actor Dependencies
+
+## Problem
+
+The Pony type system is very demanding when it comes to handling errors. The lack of `null` means that you are forced to initialize every variable and explictly handle and possible source of initialization error. In return, you get freedom from `Null Pointer Exceptions` and their equivalents. However, a naive use of Pony's `None` type when initializing dependencies can led to poor programmer ergononmics and frustration. 
+
+Take, as an example, a Pony actor that receives messages and writes them to a file. Our first naive pass might look something like:
+
+```pony
+use "files"
+
+actor Writer
+  let _file: File
+
+  new create(env: Env, file_name: String) =>
+    let auth = env.root as AmbientAuth
+    _file = File(FilePath(auth, file_name))
+
+  be record(it: String) =>
+    _file.write(it)
+```
+
+We've already hit our first problem. Our above code won't compile. Why? Well, it doesn't handle errors. For starters, `env.root as AmbientAuth` can fail. We might not be able to get the capability to acquire ambient authority. [Capabilities](https://en.wikipedia.org/wiki/Capability-based_addressing) as implemented in Pony allow you to restrict what subsystems in your program can do. Whereas in a non-capabilities language like Java, any class can open a file or a socket, in Pony you have to be given the capability to do that. This means that you might lack that capability and it could fail. If we were to address that, we would also need to address that our `File` object might not be able to be initialized. In order to deal with our errors, we'll need to make `file` be of type `(File | None)`. This union type states that we can have a file or nothing. An iteration to address this gets us almost all the way to being able to compile but not quite:
+
+```pony
+use "files"
+
+actor Writer
+  var _file: (File | None) = None
+
+  new create(env: Env, file_name: String) =>
+    try
+      let auth = env.root as AmbientAuth
+      _file = File(FilePath(auth, file_name))
+    end
+
+  be record(it: String) =>
+    _file.write(it)
+```
+
+We're now left with one more compiler error to address.
+
+```bash
```

Comment:
Note that the `bash` specifier here makes the GitHub highlighting freak out about what's seen as an opening single-quote in `couldn't` with no terminating single-quote.